openedx/adg/lms/webinars/tasks.py

```diff
@@ -0,0 +1,33 @@
+"""
+Tasks for webinars
+"""
+from datetime import datetime, timedelta
+
+from celery.task import task
+
+from openedx.adg.lms.webinars.helpers import cancel_webinar_reminders, reschedule_webinar_reminders
+from openedx.adg.lms.webinars.models import WebinarRegistration
+
+
+@task()
+def task_reschedule_webinar_reminders(webinar_id, new_start_time):
+    """
+    Reschedules all the webinar reminders.
+
+    Args:
+        webinar_id (int): Webinar Id for which reminders will be rescheduled.
+        new_start_time (str): String containing time in format ("%m/%d/%Y, %H:%M:%S")
+    """
+    new_start_time = datetime.strptime(new_start_time, '%m/%d/%Y, %H:%M:%S')
+
+    week_before_start_time = new_start_time - timedelta(days=7)
```

Comment:
There is an edge-case, but i am sure, we need to deal it with or not.

If I have created a webinar 10 days from now for at 11 pm. On the 5th day, I update its time from 11 pm to 10 pm. Its user will not get one week reminder email

Reply:
Yes, In that case I am cancelling the scheduled email, Because the email says "Webinar is one week away" and that is invalid in this case so I am skipping that reminder.